packages/wdio-allure-reporter/src/index.js

```diff
@@ -444,7 +444,8 @@ class AllureReporter extends WDIOReporter {
     }
 
     dumpJSON(name, json) {
-        this.allure.addAttachment(name, JSON.stringify(json, null, 2), 'application/json')
+        const content = JSON.stringify(json, null, 2)
+        this.allure.addAttachment(name, typeof content === 'string' ? content : `${content}`, 'application/json')
```

Comment:
Why do we need to check if `content` is a string if we use `JSON.stringify` above? That method always returns a string.

Reply:
It doesn't, it can return undefined that causes failures

Reply:
This would mean we would add `undefined` as content. How about we have a:

```js
if (!json) {
    return
}
```
instead?

Reply:
not sure, I'd always add whatever the user wants, even if it's undefined.
I'd like to see undefined in attachments if I passed it

Reply:
but undefined is not really saying anything, and it is also not a `'application/json'` mime type. Can we have something like "no content" or something like that?

Reply:
yeah, I'll set it to text plain

Reply:
but I'd not change undefined to anything, because it can be what user has passed, ex
```
let a
if (b) { a = b }
addAttachment('a', a) // I expect to see either undefined or b
```